benchmark/single-source/StringRepeating.swift

```diff
@@ -0,0 +1,143 @@
+//===--- StringRepeating.swift -------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import TestsUtils
+
+public let StringRepeating = [
+  BenchmarkInfo(name: "StringRepeating/SingleAsciiCharacterCount10",
+                runFunction: run_singleAsciiCharacterCount10,
+                tags: [.validation, .api, .String]),
+  BenchmarkInfo(name: "StringRepeating/SingleAsciiCharacterCount1",
+                runFunction: run_singleAsciiCharacterCount1,
+                tags: [.validation, .api, .String]),
+  BenchmarkInfo(name: "StringRepeating/EmptyStringCount10",
+                runFunction: run_emptyStringCount10,
+                tags: [.validation, .api, .String]),
+  BenchmarkInfo(name: "StringRepeating/SingleAsciiCharacterCount0",
+                runFunction: run_singleAsciiCharacterCount0,
+                tags: [.validation, .api, .String]),
+  BenchmarkInfo(name: "StringRepeating/26AsciiCharactersCount2",
+                runFunction: run_26AsciiCharactersCount2,
+                tags: [.validation, .api, .String]),
+  BenchmarkInfo(name: "StringRepeating/SingleCyrilicCharacterCount5",
+                runFunction: run_singleCyrilicCharacterCount5,
+                tags: [.validation, .api, .String]),
+  BenchmarkInfo(name: "StringRepeating/33CyrilicCharactersCount2",
+                runFunction: run_33CyrilicCharactersCount2,
+                tags: [.validation, .api, .String]),
+  BenchmarkInfo(name: "StringRepeating/U1F1F8U1F1FACount2",
+                runFunction: run_U1F1F8U1F1FACount2,
+                tags: [.validation, .api, .String]),
+  BenchmarkInfo(name: "StringRepeating/U301cafeCount5",
+                runFunction: run_U301cafeCount5,
+                tags: [.validation, .api, .String]),
+  BenchmarkInfo(name: "StringRepeating/LongMixedStringCount100",
+                runFunction: run_longMixedStringCount100,
+                tags: [.validation, .api, .String])
+]
```

Comment:
What is the rationale for this set of cases? Some seem redundant, and I would think we'd be more interested in small vs non-small forms

Reply:
reduced number of benchmarks to 4